File: board/data/src/main/java/com/uniboard/board/data/AllBoardsRepositoryImpl.java
package com.uniboard.board.data;
import com.mongodb.client.model.UpdateOptions;
import com.mongodb.client.model.Updates;
import com.mongodb.client.*;
import org.bson.Document;
import org.bson.conversions.Bson;
import java.util.UUID;


public class AllBoardsRepositoryImpl implements com.uniboard.board.domain.AllBoardsRepository {
        private String nameDatabase = "CreatedBoards";
        private MongoClient mongoClient;
        private Document query = new Document("isSettings", true);

        private static boolean hasCollection(final MongoDatabase db, final String collectionName) {
            assert db != null;
            assert collectionName != null && !collectionName.isEmpty();
            try (final MongoCursor<String> cursor = db.listCollectionNames().iterator()) {
                while (cursor.hasNext())
                    if (cursor.next().equals(collectionName))
                        return true;
            }
            return false;
        }

        public AllBoardsRepositoryImpl(MongoClient client) {
            mongoClient = client;
        }

        private UUID getFreeId() {
            return UUID.randomUUID();
        }


    public String add(){
        MongoDatabase database = mongoClient.getDatabase(nameDatabase);
        String uuid = getFreeId().toString();
        database.createCollection(uuid);
        edit(uuid, "");
        return uuid;

    }

    public void delete(String idBoard) {
        MongoDatabase database = mongoClient.getDatabase(nameDatabase);
        MongoCollection todoCollection = database.getCollection(idBoard);
        todoCollection.drop();
    }
    public boolean exists(String idBoard) {
        MongoDatabase database = mongoClient.getDatabase(nameDatabase);
        return hasCollection(database, idBoard);

    }
    public String settings(String id){
        MongoDatabase database = mongoClient.getDatabase(nameDatabase);
        MongoCollection todoCollection = database.getCollection(id);
        Document h = (Document) todoCollection.find(new Document("isSettings", true)).first();
        return (String) h.get("settings");
    }

    public void edit(String id, String settings){
        MongoDatabase database = mongoClient.getDatabase(nameDatabase);
        MongoCollection todoCollection = database.getCollection(id);
        Bson updates = Updates.set("settings", settings);
        UpdateOptions options = new UpdateOptions().upsert(true);
        todoCollection.updateOne(query, updates, options);
    }
}

File: board/data/src/main/kotlin/com/uniboard/board/data/AllBoardsRepositoryImpl.kt
package com.uniboard.board.data

import com.mongodb.client.MongoClient
import com.mongodb.client.MongoCollection
import com.mongodb.client.MongoDatabase
import com.mongodb.client.model.UpdateOptions
import com.mongodb.client.model.Updates
import com.uniboard.board.domain.AllBoardsRepository
import org.bson.Document
import java.util.*

class AllBoardsRepositoryImpl(private val mongoClient: MongoClient) : AllBoardsRepository {
    private val nameDatabase = "CreatedBoards"
    private val query = Document("isSettings", true)

    private fun hasCollection(db: MongoDatabase, collectionName: String?): Boolean {
        assert(!collectionName.isNullOrEmpty())
        db.listCollectionNames().iterator().use { cursor ->
            while (cursor.hasNext()) if (cursor.next() == collectionName) return true
        }
        return false
    }

    private fun getFreeId(): UUID {
        return UUID.randomUUID()
    }


    override fun add(): String {
        val database: MongoDatabase = mongoClient.getDatabase(nameDatabase)
        val uuid = getFreeId()
        database.createCollection(uuid.toString())
        return uuid.toString()
    }

    override fun delete(id: String) {
        val database: MongoDatabase = mongoClient.getDatabase("CreatedBoards")
        val todoCollection: MongoCollection<*> = database.getCollection(id)
        todoCollection.drop()
    }

    override fun exists(id: String): Boolean {
        val database: MongoDatabase = mongoClient.getDatabase("CreatedBoards")
        return hasCollection(database, id)
    }

    override fun settings(id: String): String {
        val database: MongoDatabase = mongoClient.getDatabase("CreatedBoards")
        val todoCollection: MongoCollection<*> = database.getCollection(id)
        val h = todoCollection.find(Document("isSettings", true)).first() as Document
        return h["settings"].toString()
    }

    override fun edit(id: String, settings: String) {
        val database: MongoDatabase = mongoClient.getDatabase("CreatedBoards")
        val todoCollection: MongoCollection<*> = database.getCollection(id)
        val updates = Updates.set("settings", settings)
        val options = UpdateOptions().upsert(true)
        todoCollection.updateOne(query, updates, options)
    }
}